### ISSUE TITLE
Search results hold 404 results when selecting different lib version

### ISSUE THREAD

**Issue by vitality82** [mui/mui-x#14855]
### Search keywords

docs 404

### Latest version

- [x] I have tested the latest version

### Affected products

Data Grid

### Steps to reproduce

Guys sorry for dropping in like this, I have no time for all the details of confirming to issue protocol. The fact is this:

![Image](https://github.com/user-attachments/assets/e798b367-86b0-4955-a408-78c959b0f481)

Search results only provide correct results with links if using the latest version. We are still on 6 so naturally choose v6 and all the search results are NOOP.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by michelengelen** [comment]
The only broken links I could find are those targeting the FAQ page which is not backported to v6 apparently. Related PR: https://github.com/mui/mui-x/pull/11686

**Comment by michelengelen** [comment]
cc @MBilalShafi

**Comment by MBilalShafi** [comment]
I agree that porting it back to v6 would solve the issue for this particular page and v6, it won't be solved completely.
For example, the same is reproducible in v5.

For a proper fix, and avoiding similar issue to occur in future, we need to investigate more on how to optimize algolia to only index the pages available on currently accessed major version.

**Comment by alexfauquette** [comment]
@MBilalShafi That's a global issue. We have a unique algolia crawler that runs on `https://mui.com` and `v5.mui.com`, `v6.mui.com` and other versions call the same index.

We can handle version as different instance of the docs and so have on index per version as mention in [algolia docs](https://www.algolia.com/doc/guides/sending-and-managing-data/manage-indices-and-apps/manage-your-apps/)

I'm running the creation of a v5 docs for test purpose

**Comment by vitality82** [comment]
I hope my issue has not induced noise, for the record we would stay a happy customer even if this was not solved. The ultimate goal is to find time and space to upgrade to latest version (business goals always come first). This is just one of those issues in the moment when you search for docs for a required version.

You guys rock!

**Comment by oliviertassinari** [comment]
It's the same for Material UI v5. The search is broken.

**Comment by alexfauquette** [comment]
Duplicating the crawler, and using different indexes is the solution we started to go for.

I considered 1 crawler  + index per version such that on index correspond to one sub-domain. For example, having `v5.mui.com` indexed in `v5_material-ui`.

> And done?

For the search yes, for maintaining a coherent docs for previous version not entirely.

The main issue is that we have version missmatch between core and X.

So the core docs were stable (v5) while the X one moved throw v5, v6 (refactoring the pickers), and v7 (introducing charts and tree view)

So when we archive the core docs (v5.mui.com) it matches with the latest version of the X docs (v7.mui.com/x/) which leads to 2 issues:

1. We don't know what users expect. If they are on v5.mui.com, do they want to see the v5, v6 or v7 of the X docs ?
2. We have 404 links because v5.mui.com has links to recent features (ie: https://v5.mui.com/x/react-charts/)
  ![image](https://github.com/user-attachments/assets/9af4bbbd-0035-4b8d-8b2a-2bab3c8b801c)


A solution could be to replace relative links from `v5.mui.com` to X docs by absolute links to `v7.mui.com/x/` the latest compatible version. Such that we remove 404. And if users prefer an older version, it's up to them to change it. It will stay coherent as long as they stays in the X pages.

That's an argument in favor of having sync major between core and X. Or to have a website per project

**Comment by LukasTy** [comment]
> A solution could be to replace relative links from `v5.mui.com` to X docs by absolute links to `v7.mui.com/x/` the latest compatible version. Such that we remove 404. And if users prefer an older version, it's up to them to change it. It will stay coherent as long as they stays in the X pages.

IMHO, this idea sounds like a nice compromise, at least until we align versions. 🤔 
I'm all for a superb docs experience, where everything is proper version aware, but we can also explore such a solution until we are at least version synced and then implement. 🤷

**Comment by oliviertassinari** [comment]
> Duplicating the crawler, and using different indexes is the solution we started to go for.

@alexfauquette If we want to do the least amount of work for this step, we could use the same index, docs-indra didn't change between those Material UI and MUI X majors, but it works too.

> I considered 1 crawler + index per version such that on index correspond to one sub-domain. For example, having v5.mui.com indexed in v5_material-ui.

If the index is a projet specific, it will break the cross projects search experience, so I don't think it works. Right now, that cross projet search experience on the latest major is blends the project too much, we should move the projet higher in the the Algolia tie-breaking order.

> We don't know what users expect. If they are on v5.mui.com, do they want to see the v5, v6 or v7 of the X docs ?

It should show the results of the most recent major compatible. It's achievable using or/and conditions in https://github.com/mui/material-ui/blob/1719ed4ed17a4eddc3d9ff493e6b8f5dd0bbda51/docs/src/modules/components/AppSearch.js#L452

> We have 404 links because v5.mui.com has links to recent features (ie: https://v5.mui.com/x/react-charts/)

Yeah, this was the rabbit hole. I think it's step 3.
